src/firmware/app/control/trajectory_planner.h

```diff
@@ -1,303 +1,190 @@
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS elements
+ * @param path_parameters [in/out] The path parameters that define the trajectory. The
+ * input path_parameters.num_elements defines the number of constant-parameterization
+ * elements, and the returned path_parameters.num_elements defines the number of
+ * constant-period elements in the trajectory,
  *
- * @param num_segments [in] The number of segments(elements) in the
- * max_allowable_speed_profile input array
- * @param velocity_profile [out] The array that the forwards continuous array will be
- * copied into
- * @param max_allowable_speed_profile [in] The pre-allocated array that will contain all
- * of the maximum allowable velocity values for each point on the curve
- * @param arc_segment_length [in] The length of each arc length segment in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point in
- * m/s^2
- * @param max_allowable_speed [in] The max allowable speed at any point in m/s
+ * @param constant_period_trajectory [out] A constant period trajectory
+ * generated by the specified path parameters.
  *
- * */
-void app_trajectory_planner_generateForwardsContinuousVelocityProfile(
-    const unsigned int num_segments,
-    float velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const float arc_segment_length, const float max_allowable_acceleration,
-    const float max_allowable_speed);
-
-/**
- * Modifies a forwards continuous velocity profile to also be backwards continuous based
- * on the input parameters.
- *
- * NOTE: This function does not break forwards continuity of the trajectory
- *
- * @pre The input velocity_profile is required to be FORWARDS CONTINUOUS before it is used
- * as an input to this function
- * @pre forwards_continuous_velocity_profile is pre-allocated up to at least
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
- *
- * @param num_segments [in] The number of segments(elements) in the velocity_profile array
- * @param forwards_continuous_velocity_profile [in/out] The forwards continuous velocity
- * profile to ebe modified in-place into a profile that is also backwards continuous
- * @param arc_segment_length [in] The arc segment length of each segment in the path the
- * profile is being generated for in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point on
- * the velocity profile
+ * @return A status indicating whether or not generation was successful
  */
-void app_trajectory_planner_generateBackwardsContinuousVelocityProfile(
-    const unsigned int num_segments,
-    float forwards_continuous_velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const float arc_segment_length, const float max_allowable_acceleration);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_generateConstantInterpolationPeriodPositionTrajectory(
```

Comment:
for consistency, could we rename to `app_trajectory_planner_generateConstantPeriodPositionTrajectory`?

Reply:
Renamed 👍